src/box/lua/connpool.lua

```diff
@@ -125,6 +125,42 @@ local function connect_to_candidates(candidates)
     return connected_candidates
 end
 
+local acquire_any_cond = nil
+local function acquire_any_candidate(instance_names, opts)
+    assert(type(opts) == 'table' and opts.mode)
+    assert(opts.mode == 'ro' or opts.mode == 'rw' or opts.mode =='alive')
+    local time_connect_end = clock.monotonic() + WATCHER_TIMEOUT
+
+    if not acquire_any_cond then
+        acquire_any_cond = fiber.cond()
+    end
+
+    for _, instance_name in pairs(instance_names) do
+        local ok, conn = pcall(connect, instance_name, {
+            wait_connected = false,
+            connect_timeout = WATCHER_TIMEOUT
+        })
+
+        if ok then
+            conn:on_connect(function()
+                acquire_any_cond:broadcast()
+            end)
+        end
+    end
+
+    while clock.monotonic() < time_connect_end do
+        for _, instance_name in ipairs(instance_names) do
+            local conn = connections[instance_name]
+            if is_connection_valid(conn, {}) and
+               (opts.mode == 'alive' or opts.mode == conn:mode()) then
+                return conn
```

Comment:
We generally return connections that're already have `mode`: we have to do it if we're going to use dynamic filter later. We don't do it here for the `opts.mode == 'alive'` case.

This function is specific: its only caller doesn't use the `mode` information. I think we need to explicitly clarify this nuance.

Reply:
Oh, I see. Thank you. Probably, it's guaranteed to be available in some obvious way. Though, I didn't manage to make sure. Added the comments describing the behaviour of the `acquire_***` methods and the corresponding note for specifying possible absence of the `mode()` method.